client/requestUtil.js

```diff
@@ -211,11 +212,50 @@ RequestUtil.prototype.list = function (category, startAt, maxRecords, platform)
     }
 
     return s3Helper.listNotifications(this.sqs, notificationParams, category,
-      `${this.apiVersion}/${encodeURIComponent(this.userId)}/${category}`, platform)
+      `${this.apiVersion}/${this.userIdEncoded}/${category}`)
   })
 }
 
 /**
+ * Checks do we need to use s3 list Object or SQS notifications
+ * @param {number=} startAt return objects with timestamp >= startAt (e.g. 1482435340). Could be seconds or milliseconds
+ * @param {string} category - the category ID
+ * @returns {boolean}
+*/
+RequestUtil.prototype.shouldListObject = function (startAt, category) {
+  let currentTime = new Date().getTime()
+  let startAtToCheck = this.normalizeTimestampToMs(startAt, currentTime)
+
+  return !startAtToCheck ||
+      (currentTime - startAtToCheck) > parseInt(s3Helper.SQS_RETENTION, 10) * 1000 ||
+      category !== proto.categories.BOOKMARKS ||
+      this.listInProgress === false
+}
+
+/**
+ * Checks do we need to use s3 list Object or SQS notifications
+ * @param {number=} startAt return objects with timestamp >= startAt (e.g. 1482435340). Could be seconds or milliseconds
+ * @param {number=} currentTime currentTime in milliseconds
+ * @returns {number=}
+*/
+RequestUtil.prototype.normalizeTimestampToMs = function (startAt, currentTime) {
+  let startAtToCheck = startAt
+  if (startAtToCheck && currentTime.toString().length - startAtToCheck.toString().length >= 3) {
+    startAtToCheck *= 1000
+  }
+
+  return startAtToCheck
+}
+
+/**
+ * Sets if the initial sync done or not
+ * @param {boolean}
+ */
+RequestUtil.prototype.setInitialSyncDone = function (initialSyncDone) {
```

Comment:
should we change this to `setListInProgress` and then use the inverse of `initialSyncDone` values?